common/service/ipmi/ipmi.c

```diff
@@ -29,6 +29,31 @@ K_KERNEL_STACK_MEMBER(IPMI_thread_stack, IPMI_THREAD_STACK_SIZE);
 char __aligned(4) ipmi_msgq_buffer[IPMI_BUF_LEN * sizeof(struct ipmi_msg_cfg)];
 struct k_msgq ipmi_msgq;
 
+static uint32_t iana_list[] = {
+	IANA_ID,
+#ifdef IANA_ID2
+	IANA_ID2,
+#endif
+};
+
+static int8_t find_iana_index(uint8_t *iana_buf)
+{
+	CHECK_NULL_ARG_WITH_RETURN(iana_buf, -1);
+
+	for (uint8_t iana_idx = 0; iana_idx < ARRAY_SIZE(iana_list); iana_idx++) {
+		uint8_t i;
+		for (i = 0; i < 3; i++) {
+			if (iana_buf[i] != ((iana_list[iana_idx] >> (i * 8)) & 0xFF))
+				break;
+		}
+
+		if (i == 3)
+			return iana_idx;
+	}
+
+	return -1;
+}
```

Comment:
Currently, the IANA_ID2 is only used in VF, but the modification can be reserved for other projects in the future if they also need to support another IANA.

Reply:
I think most projects will not use multiple IANAs, we'd like to avoid that. So I'd prefer to have it as a __weak function that could be overridden for any project that needs it.